hyperspy/signal.py

```diff
@@ -2784,23 +2784,25 @@ def plot(self, navigator="auto", axes_manager=None, plot_markers=True,
         elif self.tmp_parameters.has_item('filename'):
             self._plot.signal_title = self.tmp_parameters.filename
 
+        def sum_wrapper(s, axis):
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
+                    "ignore", category=UserWarning, module='hyperspy'
+                    )
+                return s.sum(axis)
```

Comment:
Is there any use for a generic warning suppressing function?  We could create a function that wraps some other function or be called explicitly. 

```python 

def suppress_warnings(func):
    def wrap(*args, **kwargs):
        with warnings.catch_warnings():
                warnings.filterwarnings(
                    "ignore", category=UserWarning, module='hyperspy'
                    )
                 return func(*args, **kwargs)
```

Maybe that is overkill though for a easy onetime fix.

Reply:
Yes, this is a good idea, but  since this is used only in one place, I would prefer to keep it as it is. It is rare that we have such a situation: a call function in hyperspy which raise a warning by hyperspy!

Reply:
Sounds good to me!